File: Java8-JavaFX/src/sevenguis/counter/Counter.java
package sevenguis.counter;

import javafx.application.Application;
import javafx.geometry.Insets;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.TextField;
import javafx.scene.layout.HBox;
import javafx.stage.Stage;

public class Counter extends Application {

    public void start(Stage stage) {
        TextField count = new TextField("0");
        count.setEditable(false);
        count.setPrefWidth(50);
        Button countUp = new Button("Count");

        countUp.setOnAction(e ->
                count.setText(1+Integer.parseInt(count.getText())+""));

        HBox root = new HBox(10, count, countUp);
        root.setPadding(new Insets(10));

        stage.setScene(new Scene(root));
        stage.setTitle("Counter");
        stage.show();
    }

    public static void main(String[] args) {
        launch(args);
    }
}

File: Scala-ScalaFX/src/main/scala/sevenguis/counter/Counter.scala
package sevenguis.counter

import scalafx.application.JFXApp
import scalafx.application.JFXApp.PrimaryStage
import scalafx.scene.Scene
import scalafx.scene.control.{Button, TextField}
import scalafx.scene.layout.HBox
import scalafx.geometry.Insets
import scalafx.event.ActionEvent
import scalafx.Includes._

object Counter extends JFXApp {
  val count = new TextField {
    text      = "0"
    editable  = false
    prefWidth = 50
  }
  val countUp = new Button("Count")

  countUp.onAction = (event: ActionEvent) => {
    count.text = (1 + count.text().toInt).toString
  }

  stage = new PrimaryStage {
    title = "Counter"
    scene = new Scene {
      content = new HBox(10) {
        padding = Insets(10)
        content = Seq(count, countUp)
      }
    }
  }
}
